includes/class-sensei-frontend.php

```diff
@@ -32,80 +58,72 @@ public function __construct () {
 				'frameborder'     => array(),
 				'allowfullscreen' => array(),
 			),
-			'video'  => Sensei_Wp_Kses::get_video_html_tag_allowed_attributes()
+			'video'  => Sensei_Wp_Kses::get_video_html_tag_allowed_attributes(),
 		);
 
-		// Template output actions
+		// Template output actions.
 		add_action( 'sensei_before_main_content', array( $this, 'sensei_output_content_wrapper' ), 10 );
 		add_action( 'sensei_after_main_content', array( $this, 'sensei_output_content_wrapper_end' ), 10 );
 		add_action( 'sensei_lesson_archive_lesson_title', array( $this, 'sensei_lesson_archive_lesson_title' ), 10 );
-
-		// 1.2.1
-		add_action( 'wp_head', array( $this, 'sensei_complete_lesson' ), 10 );
+		add_action( 'init', array( $this, 'sensei_complete_lesson' ), 10 );
```

Comment:
You could also hook into wp action and take advantage of the loop (instead of passing lesson_id).